var/spack/repos/builtin/packages/ginac/package.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Ginac(CMakePackage):
+    """GiNaC is a C++ library. It is designed to allow the creation of
+    integrated systems that embed symbolic manipulations together with more
+    established areas of computer science (like computation-intense numeric
+    applications, graphical interfaces, etc.) under one roof."""
+
+    homepage = "https://www.ginac.de/"
+    url      = "https://www.ginac.de/ginac-1.7.8.tar.bz2"
+    git      = "git://www.ginac.de/ginac.git"
+
+    maintainers = ['jppelteret']
+
+    version('1.7.8', sha256='0c86501aa6c72efd5937fce42c5e983fc9f05dadb65b4ebdb51ee95c9f6a7067')
+
+    variant('ginsh', default=True, description='Build ginsh, the GiNaC interactive shell')
+
+    depends_on('cmake@2.8:',    type='build')
+
+    depends_on('cln')
+    depends_on('python@3:')
+    depends_on('bison@2.3:',    when='+ginsh')
+    depends_on('flex@2.5.33:',  when='+ginsh')
```

Comment:
These two are often type=build

Reply:
This `ginsh` terminal application does parsing of user input, so I believe these are run-time dependencies and not just build dependencies.

Reply:
In that case should they be type=build/run?

Reply:
I'd be happy to make the change, but I'm curious: How is this not implied, in the same way that I don't specify build+run for `depends_on('cln')`?

Reply:
The default deptype is build/link, not build/run.

Reply:
Wow, I really dropped the ball on this @adamjstewart. Sorry for taking so long to reply.

> This ginsh terminal application does parsing of user input, so I believe these are run-time dependencies and not just build dependencies.

I've just confirmed that the terminal application works just fine as is, so I must have been mistaken, and these are `build/link` dependencies. So I'm pretty sure that everything is fine as is. They have recently released a new version though, so I'll quickly add that and check that it works too.

Reply:
It might only work because you have flex/bison installed on the system and it picks them up at run-time. But most of the time, build-only is fine.

Reply:
I've circled around and I think that you were right the first time. I've now set the flags to be `build` only, as you previously suggested. I've taken another look at their documentation and their CMake configuration files, and I've convinced myself that this is appropriate. Thank you for being patient while I worked this out @adamjstewart.